facets_mcp/utils/file_utils.py

```
@@ -72,13 +77,14 @@ def get_file_content(file_path: str) -> str:
                 with open(file_path, 'r', encoding='cp1252') as f:
                     return f.read()
             except UnicodeDecodeError:
-                error_message = f"Could not read file with any supported encoding: {str(e)}"
-                return error_message
+                raise UnicodeDecodeError(
+                    e.encoding, e.object, e.start, e.end,
+                    f"Could not read file {file_path} with any supported encoding"
+                )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)